### PR TITLE
Site Settings in SwiftUI: Privacy

### PR DESCRIPTION
**Pull request by kean**
**HACK week:** update Site Settings to use SwiftUI

As I was working on the [original PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/20874), I realized two things:

- `SiteSettingsViewController.m` has a few extensions and much more code than I initially estimated. It's not feasible to rewrite the whole thing within a week.
- Most of the cells in `SiteSettingsViewController` show custom UIKit-based screen. You can't really use any of the native SwiftUI form components. There is too much UIKit-SwiftUI dance needed to rewrite this screen without rewriting the leaf screens first.

So I decided to salvage what's left from the [original PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/20874) and pivot to replacing the "leaf" screens and adding all the reusable components and other infra for SwiftUI.

Changes:
- Add `SettingsPicker` to replace `SettingsSelectionViewController` (`SwiftUI.Picker` has a non-customizable design that can't be changed to match the WP design)
- Update Site Settings to use the new picker
- Rewrite SiteVisibilityHelpers in Swift

## To test:

- Open Dashboard / Menu / Site Settings / Privacy
- Select a new value
- Verify that the settings is updated (check on the web or on another phone)
- Verify that hint (section footer) is updated
- Return to Site Settings
- Verify that Privacy value is updated

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/101070cd-d616-4a98-9fac-961e43f544b1

## Regression Notes
1. Potential unintended areas of impact: Site Settings / Privacy
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual test and updated existing unit tests
3. What automated tests I added (or what prevented me from doing so): test eligible privacy settings

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController+Swift.swift
Comment (kean) [inline WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController+Swift.swift:24]: It's a bit awkward to use any SwiftUI view that accepts `Binding` from `UIKit` because just binding itself doesn't update the view.

File: WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController+Swift.swift
Comment (mokagio) [inline WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController+Swift.swift:409]: Thanks for the reverse DNS key usage 👍

File: WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:63]: [Q] Can you explain why you're setting the content shape? I did a quick search and it seems like its used to configure the tappable area of a view, but I was wondering why this is needed when the parent view is a Button. Thanks!

File: WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:3]: Neat ✨

File: WordPress/WordPressTest/SiteVisibilityTests.swift
Comment (momo-ozawa) [inline WordPress/WordPressTest/SiteVisibilityTests.swift]: Typo?
```suggestion
        // Then private is eligible
```

**Review by momo-ozawa — COMMENTED** [review]
Nice! Works as described, left a question. I looked at the SwiftUI code and while it made sense to me, I'll defer to @alpavanoglu (or someone else with more SwiftUI experience) to approve the PR 🙇‍♀️

**Review by alpavanoglu — APPROVED** [review]
LGTM! There's a couple of questions by Momo. I'm approving so if that's clarified, we can merge.

File: WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift
Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:63]: It has to do with hit testing. I'm not sure if there is any good documentation on it. But if you use `.plain` buttons in ` List`, it uses only primitive leaf views for hit testing, not the containers like `HStack`. The `contentShape` modifier changes the hit test area.

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:63]: Gotcha, thanks for the explanation

Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:63]: This is probably the only scenario where you need it.

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Settings/SettingsPicker.swift:63]: TIL 🙏

**Pull request merged**
